### DISCUSSION TITLE
Creating a new "@sablier/test-utils" package

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#887] [Ideas]
This would be a separate repository and npm package.

Things we could throw in there:

- Most of the things under `./test/utils`
- The test mocks under `./test/mocks` in V2 Core
- `BaseScript`, which is [95% the same](https://www.diffchecker.com/FHj3rjHI/) in V2 Core and V2 Periphery

RFC @sablier-labs/solidity

**Comment by smol-ninja** [comment]
What files do you think we can move to a separate repo?

Most of the contracts under `./test/utils` import from `src` dir. Similarly, except for ERC20 mocks, all the mock contracts import from `src` dir. Until we only move contracts that are independent of core, it would create cross-dependency. And if we move only the independent part, we will still need some of the utils and mocks in core contracts.

**Reply by PaulRBerg** [reply]
Good point.

It looks like there's only the `BaseScript` and the mocks that could be DRY-ified.

But creating a bespoke repo just for them would be overkill.

**Reply by smol-ninja** [reply]
`BaseScript` can be reused in periphery. Check [PR](https://github.com/sablier-labs/v2-periphery/pull/315) and please feel free to merge if you like it. However, for this to work, we will have to include `BaseScript` in the npm package as well.

In periphery, we only have `ERC20` mock which we can also reuse from core contracts but I think in the future, we are more likely going to use mocks from OZ or forge so either we can leave mock in periphery or ofc we can reuse from core. Wdyt?

**Reply by PaulRBerg** [reply]
Thanks for creating that PR, but I closed it because I would prefer the `BaseScript` not to be included in the npm package. That script relies on certain third-party components like Sphinx and Forge Std, which we would have to also include in the npm package, and I would rather not want us to have to do this (consider the maintenance cost).

Regarding the mocks - yeah, I agree, let's keep them duplicated for now.
